commit e4a75172f83894a4e821807b6d94770095ddaeff
Author: Erik de Castro Lopo <erikd@mega-nerd.com>
Date:   Mon Nov 9 22:10:35 2015 +1100

Index: libsndfile-1.0.21/src/common.c
===================================================================
--- libsndfile-1.0.21.orig/src/common.c	2015-11-30 11:28:03.000000000 +0100
+++ libsndfile-1.0.21/src/common.c	2015-11-30 11:28:03.000000000 +0100
@@ -805,7 +805,8 @@
 	if (psf->headindex + bytes > SIGNED_SIZEOF (psf->header))
 	{	int most ;
 
-		most = SIGNED_SIZEOF (psf->header) - psf->headindex ;
+		most = SIGNED_SIZEOF (psf->header) - psf->headend ;
+
 		psf_fread (psf->header + psf->headend, 1, most, psf) ;
 		memset ((char *) ptr + most, 0, bytes - most) ;
 
